ibis/expr/types/structs.py

```diff
@@ -200,13 +200,13 @@ def __getitem__(self, name: str) -> ir.Value:
         │ NULL   │
         │ NULL   │
         └────────┘
-        >>> t.s["foo_bar"]
+        >>> t.s["foo_bar"]  # doctest: +ELLIPSIS
         Traceback (most recent call last):
-            ...
-        KeyError: 'foo_bar'
+        ...
+        ibis.common.exceptions.FieldsNotFoundError: 'foo_bar' not found in StructColumn object. Possible options: ('a', 'b')
         """
         if name not in self.names:
-            raise KeyError(name)
+            raise FieldsNotFoundError(self, name, self.names)
```

Comment:
We didn't have a test for` Struct.__getitem` before, but if we had, then we would see that this change is breaking. Before `Struct["foo_bar"]` raised a KeyError, now it raises a FieldsNotFoundError. I think something that derives from AttributeError is more semantically correct: we know what fields are present at compile time, whereas I think of KeyError I think of for when you have a runtime-missing key. IDK if this is standard.

Reply:
EDIT: now FieldsNotFoundError derives ONLY from IbisError, and neither KeyError nor AttributeError.